Content.Server/HeadSlime/HeadSlimeSystem.Transform.cs

```diff
@@ -0,0 +1,257 @@
+using Content.Shared.Nutrition.Components;
+using Content.Shared.Popups;
+using Content.Shared.Roles;
+using Content.Shared.Tools;
+using Content.Shared.Tools.Components;
+using Content.Shared.HeadSlime;
+using Robust.Shared.Audio;
+using Robust.Shared.Utility;
+using Content.Server.Actions;
+using Content.Shared.Actions;
+using Robust.Shared.Prototypes;
+using Content.Shared.Actions.ActionTypes;
+using System.Numerics;
+using Robust.Shared.Map;
+using Content.Shared.Body.Systems;
+using Content.Shared.Body.Components;
+using Content.Server.Body.Components;
+using Content.Shared.Body.Organ;
+using Content.Shared.Body.Part;
+
+namespace Content.Server.HeadSlime
+{
+    /// <summary>
+    ///     Handles HeadSlime propagation and inherent HeadSlime traits
+    /// </summary>
+    public sealed partial class HeadSlimeSystem
+    {
+        [Dependency] private readonly NpcFactionSystem _faction = default!;
+        [Dependency] private readonly NPCSystem _npc = default!;
+        [Dependency] private readonly MovementSpeedModifierSystem _movementSpeedModifier = default!;
+        [Dependency] private readonly IChatManager _chatMan = default!;
+        [Dependency] private readonly MindSystem _mind = default!;
+        [Dependency] private readonly SharedAudioSystem _audio = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [Dependency] private readonly SharedBodySystem _body = default!;
+
+        /// <summary>
+        ///     This is the general purpose function to call if you want to head slime an entity.
+        ///     It handles only humanoids.
+        /// </summary>
+        /// <param name="target">the entity being Head Slimed</param>
+        /// /// /// <param name="mobState"></param>
+        
+        public void HeadSlimeEntity(EntityUid target, MobStateComponent? mobState = null, bool HeadSlimeQueen = false, bool PutHSRHatOn = false)
+        {
+            //Can't be HeadSlimed again
+            if (HasComp<HeadSlimeComponent>(target))
+                return;
+
+            if (!Resolve(target, ref mobState, logMissing: false))
+                return;
+
+            //HeadSlime's only attach to Humanoids
+            if (!HasComp<HumanoidAppearanceComponent>(target))
+                return;
+
+            //Checking if target needs a free hat slot.
+            if(!HeadSlimeQueen)
+            {
+                if(!PutHSRHatOn)
+                {
+                    if (_inventory.TryGetSlotEntity(target, "head", out var _))
+                        return;
+                }
+            }
+
+            //Welcome to the Head Slime Club
+            var HeadSlimecomp = AddComp<HeadSlimeComponent>(target);
+
+            //HeadSlimes dont need to eat or drink but TableSalt becomes lethal
+            RemComp<HungerComponent>(target);
+            RemComp<ThirstComponent>(target);
```

Comment:
Couldn't you just set the BaseDecayRate to 0 and the CurrentHunger to 100? Removing the component seems a bit hard to reverse.